### PR TITLE
Update documents for v0.28.0

### PR DESCRIPTION
**Pull request by bidoubiwa**
Related to: https://github.com/meilisearch/meilisearch/issues/2372

## All the changes:

- [x] GET /documents/:uid, GET /documents Add the possibility to reduce the body payload by using a query parameter called fields (previously called attributesToRetrieve), check the issue and the spec for the entire behavior.
- [x] Wrap the objects in a results key.
- [x] GET /documents has pagination metadata, added limit (default: 20), offset (default: 0), total.
- [x] ⚠️ The displayedAttributes settings do not impact anymore the displayed fields returned in the /documents endpoints. These settings only affect the /search endpoint.

### CODE REVIEW

File: src/documents.rs [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    /// # let MEILISEARCH_API_KEY = option_env!("MEILISEARCH_API_KEY").unwrap_or("masterKey");
+    /// #
+    /// # let client = Client::new(MEILISEARCH_HOST, MEILISEARCH_API_KEY);
+    ///
+    /// let index = client.index("documents_query_offset");
+    ///
+    /// let mut documents_query = DocumentsQuery::new(&index);
+    ///
+    /// documents_query.with_offset(1);
+    /// ```
```
Comment (alallema) [inline src/documents.rs:87]:
```suggestion
    /// let index = client.index("documents_query_limit");
    ///
    /// let mut documents_query = DocumentsQuery::new(&index);
    ///
    /// documents_query.with_limit(1);
    /// ```
```

File: src/documents.rs [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    /// # Example
+    ///
+    /// ```
+    /// # use meilisearch_sdk::{client::*, indexes::*, documents::*};
+    /// #
+    /// # let MEILISEARCH_HOST = option_env!("MEILISEARCH_HOST").unwrap_or("http://localhost:7700");
+    /// # let MEILISEARCH_API_KEY = option_env!("MEILISEARCH_API_KEY").unwrap_or("masterKey");
+    /// #
+    /// # let client = Client::new(MEILISEARCH_HOST, MEILISEARCH_API_KEY);
+    /// let index = client.index("documents_query_offset");
```
Comment (alallema) [inline src/documents.rs:104]:
```suggestion
    /// let index = client.index("documents_query_fields");
```
?

File: src/documents.rs [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    pub fn with_fields(
+        &mut self,
+        fields: impl IntoIterator<Item = &'a str>,
+    ) -> &mut DocumentsQuery<'a> {
+        self.fields = Some(fields.into_iter().collect());
+        self
+    }
+
+    /// Specify the limit.
+    ///
```
Comment (alallema) [inline src/documents.rs:119]: Don't think it's the good title

**Review by alallema — COMMENTED** [review]
Great 🔥 !!! Just some comments

File: src/indexes.rs [hunk 4]
```diff
@@ -346,34 +354,32 @@ impl Index {
...
-            url.push_str(attributes_to_retrieve);
-        }
-        request::<(), Vec<T>>(&url, &self.client.api_key, Method::Get(()), 200).await
+    ) -> Result<DocumentsResults<T>, Error> {
+        let url = format!("{}/indexes/{}/documents", self.client.host, self.uid);
+
+        request::<(), DocumentsResults<T>>(&url, &self.client.api_key, Method::Get(()), 200).await
+    }
+
+    pub async fn get_documents_with<T: DeserializeOwned + 'static>(
```
Comment (bidoubiwa) [inline src/indexes.rs:371]: missing example

File: src/documents.rs [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    /// The maximum number of documents returned.
+    /// If the value of the parameter `limit` is `n`, there will never be more than `n` documents in the response.
+    /// This is helpful for pagination.
+    ///
+    /// Example: If you don't want to get more than two documents, set limit to `2`.
+    /// Default: `20`
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub limit: Option<usize>,
+
+    pub fields: Option<Vec<&'a str>>,
```
Comment (bidoubiwa) [inline src/documents.rs:34]: missing `#[serde(skip_serializing_if = "Option::is_none")]`

**Pull request merged**